File: src/comun/componente/DiscoDuro.java
package comun.componente;

/**
 *
 * @author Barrionuevo Diego
 */
public class DiscoDuro extends Componente {

    private String fabricante;
    private String modelo;
    private String numeroSerie;
    private double capacidad;//kBytes
    private String tipoInterfaz;//(SCSI , PCI, IDE)
    private String firmware;
    private int cantidadParticiones;
    //private int bytesPorSector;//cantidad
    //(Total de sectores, cilindros, cabeceras, pistas, pistas por cilindro)

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getNumeroSerie() {
        return numeroSerie;
    }

    public void setNumeroSerie(String numeroSerie) {
        this.numeroSerie = numeroSerie;
    }

    public double getCapacidad() {
        return capacidad;
    }

    public void setCapacidad(double capacidad) {
        this.capacidad = capacidad;
    }

    public String getTipoInterfaz() {
        return tipoInterfaz;
    }

    public void setTipoInterfaz(String tipoInterfaz) {
        this.tipoInterfaz = tipoInterfaz;
    }

    public String getFirmware() {
        return firmware;
    }

    public void setFirmware(String firmware) {
        this.firmware = firmware;
    }

    public int getCantidadParticiones() {
        return cantidadParticiones;
    }

    public void setCantidadParticiones(int cantidadParticiones) {
        this.cantidadParticiones = cantidadParticiones;
    }

}

File: Cliente/Linux/componentes/discoduro.py

class DiscoDuro():
    
    def __init__(self):                            
        self.fabricante = ""
        self.modelo = ""
        self.numero_serie = ""        
        self.tipo_interfaz = ""
        self.firmware = ""
        self.cantidad_particiones = 0
        self.tamanio = ""
    
    def setid(self, id):
        self.id = id
        
    def setfabricante(self, fabricante):
        self.fabricante = fabricante

    def setmodelo(self, modelo):
        self.modelo = modelo
        
    def setnumeroserie(self, numero_serie):
        self.numero_serie = numero_serie
        
    def settamanio(self, tamanio):
        """En Bytes"""
        self.tamanio = tamanio
    
    def settipointerfaz(self, tipo_interfaz):
        """SCSI, PC, IDE"""
        self.tipo_interfaz = tipo_interfaz
        
    def setfirmware(self, firmware):
        self.firmware = firmware
        
    def setcantidadparticiones(self, cantidad_particiones):
        self.cantidad_particiones = cantidad_particiones        
    
    def tostr(self):        
        disco = 'Modelo:'+ self.modelo + '\n' +'Fabricante:'+self.fabricante+ '\n'+'NúmeroSerie:'+ self.numero_serie + '\n' + 'Capacidad:'+ str(self.tamanio)+'\n'+'TipoInterfaz:'+ self.tipo_interfaz + '\n' + 'Firmware:'+ self.firmware+'\n'+'CantidadParticiones:'+ str(self.cantidad_particiones)
        return disco
